lib/drivers/medtronic/medtronicSimulator.js

```diff
@@ -335,8 +346,8 @@ exports.make = function(config){
                   event.duration = currStatus.duration;
 
                   resumeBasal = _.clone(currBasal);
-                  resumeBasal.time = new Date(Date.parse(currStatus.time) + currStatus.duration).toISOString();
-                  resumeBasal.deviceTime = sundial.formatDeviceTime(Date.parse(currStatus.deviceTime) + currStatus.duration);
+                  resumeBasal.time = addDurationToTime(currStatus.time,currStatus.timezoneOffset, currStatus.duration).toISOString();
```

Comment:
I don't _think_ the function should be used here. The subtracting then adding back `timezoneOffset` gymnastics is specific to working with `deviceTime`s, and this one's `time`. You should be able to just add the `duration` here.